pennylane/devices/qutrit_mixed/initialize_state.py

```diff
@@ -0,0 +1,102 @@
+"""Functions to prepare a qutrit mixed state."""
+
+from typing import Iterable, Union
+import numpy as np
+import pennylane as qml
+from pennylane import (
+    QutritBasisState,
+)
+from pennylane.operation import (
+    StatePrepBase,
+    Operation,
+)
+from pennylane.wires import Wires
+
+qudit_dim = 3  # specifies qudit dimension
+
+
+def create_initial_state(
+    wires: Union[qml.wires.Wires, Iterable],
+    prep_operation: Operation = None,
+    like: str = None,
+):
+    r"""
+    Returns an initial state, defaulting to :math:`\ket{0}\bra{0}` if no state-prep operator is provided.
+
+    Args:
+        wires (Union[Wires, Iterable]): The wires to be present in the initial state
+        prep_operation (Optional[StatePrepBase]): An operation to prepare the initial state
+        like (Optional[str]): The machine learning interface used to create the initial state.
+            Defaults to None
+
+    Returns:
+        array: The initial state of a circuit
+    """
+    if isinstance(wires, Wires):
+        wire_array = wires.toarray()
+    else:
+        wire_array = np.array(wires)
+    num_wires = len(wire_array)
+
+    if not prep_operation:
+        rho = _create_basis_state(num_wires, 0)
+
+    elif isinstance(prep_operation, QutritBasisState):
+        rho = _apply_basis_state(prep_operation.parameters[0], wire_array)
+
+    elif isinstance(prep_operation, StatePrepBase):
+        rho = _apply_state_vector(
+            prep_operation.state_vector(wire_order=list(wire_array)), num_wires
+        )
```

Comment:
The helpers for creating the initial state should also accept a `like` argument so that the correct type of array is returned.

Reply:
I do that at the end of this function, would you like it to be changed so that the state is immediately returned? I originally did this to avoid repeat code.

Reply:
Nope, I missed it while skimming the code. Thanks!